src/moonlink/src/storage/iceberg/rest_catalog_test_guard.rs

```diff
@@ -34,15 +34,37 @@ impl RestCatalogTestGuard {
             None
         };
         Ok(Self {
-            namespace: ns_ident,
+            namespace: Some(ns_ident),
             table: table_ident,
         })
     }
 }
 
 impl Drop for RestCatalogTestGuard {
     fn drop(&mut self) {
-        let table = self.table.take();
+        fn drop_namespace<'a>(
```

Comment:
nit: not sure why don't we make it a separate function?
the benefit of which is less indentation